### PR TITLE
Multi-Tenant Pipeline

### PR DESCRIPTION
**Pull request by swainn**
Implements full flow for multi-tenant authentication.
* Moves `sso_tenant` view to new psa.py view module and renames it to `tenant`.
* Adds psa URLs manually so we can override with custom own versions of the views.
* Implements custom `auth` and `complete` views that check for multi-tenant conditions and adds tenant to the `backend` for use looking up tenant-specific settings later on.
* Completes implementation of `tenant` view so that it calls `do_auth` when tenant info has been set on the backend.
* Adds `SOCIAL_AUTH_FIELDS_STORED_IN_SESSION = ['tenant']` to `settings.py` so that the `tenant` field of the tenant form will be saved in the session storage. The value is retrieved in `complete` view to load the tenant settings on callback.
* Adds `MultiTenantMixin` class to add to backends so they can support multi-tenant authentication.
* Adds backend "namespace" to local storage keys so that the tenant name that is saved for one backend isn't populated in the Tenant form for another backend.

Implements client feedback
* Add setting to override the regex used to validate the tenant field (`SSO_TENANT_REGEX`)

- [x] Linted
- [x] 100% test coverage
- [x] Test cover all common cases
- [x] Doctstring on new methods and classes

### CODE REVIEW

**Comment by swainn** [comment]
Still needs tests.

File: tethys_portal/views/psa.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+def auth(request, backend):
+    # Redirect to tenant page if MULTI_TENANT setting configured for supported backend
+    if isinstance(request.backend, MultiTenantMixin) and request.backend.setting('MULTI_TENANT', None) is not None:
+        return redirect('social:tenant', backend=backend)
+    return do_auth(request.backend, redirect_name=REDIRECT_FIELD_NAME)
+
+
+@never_cache
+@csrf_exempt
+@psa('social:complete')
```
Comment (gagelarsen) [inline tethys_portal/views/psa.py:34]: PSA? what does that stand for?

**Review by gagelarsen — COMMENTED** [review]
Looks good to me. Just need to fix the failures and do a Demo.

File: tethys_portal/views/psa.py (hunk 1 above)
Comment (swainn) [reply tethys_portal/views/psa.py:34]: Python Social Auth

**Review by gagelarsen — APPROVED** [review]
Looks good!

**Pull request merged**